libs/serialization/include/hpx/any.hpp

```diff
@@ -0,0 +1,16 @@
+//  Copyright (c) 2020 ETH Zurich
+//
+//  SPDX-License-Identifier: BSL-1.0
+//  Distributed under the Boost Software License, Version 1.0. (See accompanying
+//  file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+#pragma once
+
+#include <hpx/datastructures/any.hpp>
+#include <hpx/serialization/serializable_any.hpp>
+
+namespace hpx {
+    using hpx::util::any;
+    using hpx::util::bad_any_cast;
+    using hpx::util::make_any;
```

Comment:
Should the local-only (non-serializable) versions here as well? There are also things like `unique_any`.

Reply:
That's the question. The same applies to `function` as well. @sithhell, @biddisco, @K-ballo, @aurianer preferences? Do we put only one serializable and copyable name corresponding to the standard library name in `hpx` or all the variations as well? I think we can go with all of them in `hpx` since they mostly share implementations, and it's unlikely we'd want to remove any of the variants (except perhaps change names).

Reply:
If we want to go for the 'standard' version only we should use `hpx::util::any_nonser<>`, if we decided to have all of them, well then `hpx::any` should still be an alias for the non-serializable variant, while `hpx::distributed::any` should refer to the serializable one.

Reply:
There's the same problem here as with checkpointing and resiliency. Serialization is usually used with distributed applications, but not in itself a distributed feature. There's also the option of making `hpx::any` the same as `hpx::util::any_nonser` and making `hpx::any_ser` (or `any_serializable`) the same as `hpx::util::any` i.e. flipping which variant gets the extra "annotation" in its name.

Reply:
Ohh my... so many options. Like a kid in a candy shop ;-)

Reply:
Well, yes. I'd just like to avoid having to change the names in the next release again.

Is there actually a runtime cost to using the serializable variants of `any` or `function`, or is it purely compile-time? @K-ballo?

Reply:
Should be pure compile-time costs, the only runtime cost is a slight increase in the size of the (manually constructed) vtable.

Reply:
Ok... so why don't we just do "no decision" and keep the same names as in `hpx::util` for both `any` and `function`. `tuple`, `optional` etc. are serializable as well (just nonintrusively). If we ever have something like a `HPX_WITH_SERIALIZATION` option we can make `hpx::any/function` be the nonserializable variants when the option is off.

Reply:
@msimberg let's do that. Since we can't make up our minds, leaving things as is might be the best way forward...